src/pages/FoodItem/QuantityInput/QuantityInput.tsx

```diff
@@ -1,57 +1,61 @@
-import React from "react";
-import {
-  PlusButton,
-  MinusButton,
-  PlusIcon,
-  MinusIcon,
-} from "../../../assets/svgs";
+import React, { useState } from "react";
+import { useFormContext, useController } from "react-hook-form";
+import { OperatorButton } from "./OperatorButton.style";
+import QuantityIcon from "./QuantityIcon";
 import { QuantityInputStyle } from "./QuantityInput.style";
-import { animated, useSpring } from "react-spring";
-import { OperatorDiv } from "./OperatorDiv.style";
 
 type Props = {
   quantity: number;
-  onClick: React.Dispatch<React.SetStateAction<number>>;
+  setQuantity: React.Dispatch<React.SetStateAction<number>>;
 };
 
-export default React.memo(function QuantityInput({ quantity, onClick }: Props) {
-  const clickHandler = (action: string): void => {
+export default React.memo(function QuantityInput({
+  quantity,
+  setQuantity,
+}: Props) {
+  const [disableDecrement, setDisableDecrement] = useState(true);
+  const { control } = useFormContext();
+  const {
+    field: { onChange },
+  } = useController({
+    name: `quantity`,
+    control: control,
+  });
+
+  const clickHandler = (action: "increment" | "decrement"): void => {
     switch (action) {
       case "increment":
-        onClick(quantity + 1);
-        return;
+        if (quantity === 1) setDisableDecrement(false);
+        onChange(quantity + 1); //update form state
+        setQuantity(quantity + 1); //update local state
```

Comment:
Use input elements with type="button" for the plus and minus icon. Make it controlled input with useController.
onClick, update local quantity state, update form quantity state.

Local state is used to render my quantity span. Passed into QuantityIcon component as prop. 
Form state is for form. Will be used later to calculate totalPrice.